src/main/java/mate/academy/EventManager.java

```diff
@@ -1,16 +1,23 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
```

Comment:
Using CopyOnWriteArrayList can be costly in terms of performance if the register and deregister operations are frequent, as each operation creates a new copy of the underlying array. If these operations are common, consider using alternatives like `ConcurrentHashMap.newKeySet()` or `Collections.synchronizedList(new ArrayList<>())` with manual synchronization during iterations.